pkgs/development/compilers/llvm/common/libunwind/default.nix

```
@@ -0,0 +1,86 @@
+{ lib
+, stdenv
+, release_version
+, patches ? []
+, src ? null
+, llvm_meta
+, version
+, monorepoSrc ? null
+, runCommand
+, cmake
+, ninja
+, python3
+, libcxx
+, enableShared ? !stdenv.hostPlatform.isStatic
+}:
+let
+  pname = "libunwind";
+  src' = if monorepoSrc != null then
+    runCommand "${pname}-src-${version}" {} (''
+      mkdir -p "$out"
+      cp -r ${monorepoSrc}/cmake "$out"
+      cp -r ${monorepoSrc}/${pname} "$out"
+      mkdir -p "$out/libcxx"
+      cp -r ${monorepoSrc}/libcxx/cmake "$out/libcxx"
+      cp -r ${monorepoSrc}/libcxx/utils "$out/libcxx"
+      mkdir -p "$out/llvm"
+      cp -r ${monorepoSrc}/llvm/cmake "$out/llvm"
+    '' + (lib.optionalString (lib.versionAtLeast release_version "15") ''
+      cp -r ${monorepoSrc}/llvm/utils "$out/llvm"
+    '') + ''
+      cp -r ${monorepoSrc}/runtimes "$out"
+    '') else src;
+
+  postUnpack = lib.optionalString (lib.versions.major release_version == "12") ''
+    ln -s ${libcxx.src}/libcxx .
+    ln -s ${libcxx.src}/llvm .
+  '';
+
+  prePatch = lib.optionalString (lib.versionAtLeast release_version "15" && lib.versionOlder release_version "18") ''
+    cd ../${pname}
+    chmod -R u+w .
+  '';
+
+  postPatch = lib.optionalString (lib.versionAtLeast release_version "15" && lib.versionOlder release_version "18") ''
+    cd ../runtimes
+  '';
```

I think this would be better.
```suggestion
  prePatch = lib.optionalString (builtins.length patches > 0 && lib.versionAtLeast release_version "15") ''
    cd ../${pname}
    chmod -R u+w .
  '';

  postPatch = lib.optionalString (builtins.length patches > 0 && lib.versionAtLeast release_version "15"") ''
    cd ../runtimes
  '';
```
Probably want to pull `builtins.length patches > 0` out into a `let`-bound variable too.

---

Done